Python/optimizer.c

```diff
@@ -1225,6 +1225,9 @@ int effective_trace_length(_PyUOpInstruction *buffer, int length)
 }
 #endif
 
+static int uop_optimize_initialized = 0;
+static int uop_optimize_flag = 0;
```

Comment:
This isn't thread safe or subinterpreter safe. You might want to store the value in a `PyThreadState` / `PyInterpreterState` in `pylifecycle.c` instead.

Reply:
Thanks, I don't know where to add `uop_flag_initialized` and `optimize_uop_flag` in which .h file and line (sorry) . I thought of this

```c
int uop_optimize(PyFrameObject *frame, PyObject *buffer, 
                 int curr_stackentries, int *dependencies)
{
    PyThreadState *tstate = PyThreadState_Get();
    if (tstate == NULL) {
        return -1;
    }
    
    if (!tstate->uop_flag_initialized) {
        tstate->uop_flag_initialized = 1;
        const char *env_var = Py_GETENV("PYTHON_UOPS_OPTIMIZE");
        if (env_var == NULL || *env_var == '\0' || *env_var > '0') {
            tstate->optimize_uop_flag = 1;
        }
        else {
            tstate->optimize_uop_flag = 0;
        }
    }

    if (tstate->optimize_uop_flag == 1) {
        ...
    }

    return 0;
}
```

Do I need to modify pylifecycle.c?